main.go

```diff
@@ -83,16 +105,61 @@ func parseArgsAndLoadConfig(args []string) (conf configuration, listen string, a
 	}
 
 	confListen := strings.Split(conf.Server.Listen, ":")
-	if len(confListen) > 1 && string(port) != confListen[1] {
-		listen = fmt.Sprintf("%s:%d", confListen[0], port)
-		log.Infof("Overriding port from config %s with %d from the commandline", confListen[1], port)
+	if len(confListen) > 1 && port != confListen[1] {
+		listen = fmt.Sprintf("%s:%s", confListen[0], port)
+		log.Infof("Overriding listen addr from config %s with new port from the commandline: %s", conf.Server.Listen, listen)
 	} else {
 		listen = conf.Server.Listen
 	}
 
 	return
 }
 
+func startRuntimeCollector(conf configuration) {
+	if conf.Statsd.Addr == "" {
+		return
+	}
+
+	s, err := g2s.Dial("udp", conf.Statsd.Addr)
+	if err != nil {
+		log.Fatalf("Unable to connect to Statsd on %s - %s", conf.Statsd.Addr, err)
+	}
+
+	gaugeFunc := func(key string, val uint64) {
+		s.Gauge(1.0, conf.Statsd.Namespace+key, strconv.FormatUint(val, 10))
```

Comment:
If you set the prefix per above you can get rid of the `conf.Statsd.Namespace` bit